cpp/examples/downstream/cmake/get_kvikio.cmake

```diff
@@ -0,0 +1,28 @@
+# =============================================================================
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+# Use CPM to fetch KvikIO, which makes `kvikio::kvikio` available for `target_link_libraries`
+function(find_and_configure_kvikio MIN_VERSION)
+
+  CPMAddPackage(
```

Comment:
I would use CPMFIndPackage so you search for a local version before building from source.

Reply:
Changing to `CPMFIndPackage` without doing anything else it fails:
```
(kvikio-0818) mkristensen@mkristensen-workstation:~/repos/kvikio/cpp/examples/downstream/build$ cmake ..
-- The CXX compiler identification is GNU 9.5.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/mkristensen/apps/miniconda3/envs/kvikio-0818/bin/x86_64-conda-linux-gnu-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Downloading CPM.cmake to /home/mkristensen/repos/kvikio/cpp/examples/downstream/build/cmake/CPM_0.35.5.cmake
CMake Error at build/cmake/CPM_0.35.5.cmake:287 (cpm_check_if_package_already_added):
  cpm_check_if_package_already_added Function invoked with incorrect
  arguments for function named: cpm_check_if_package_already_added
Call Stack (most recent call first):
  cmake/get_kvikio.cmake:18 (CPMFindPackage)
  cmake/get_kvikio.cmake:28 (find_and_configure_kvikio)
  CMakeLists.txt:27 (include)
```

Reply:
`CPMFindPackage` doesn't support the short syntax form you are using with `CPMAddPackage`. So this is the change you need:

```cmake
  CPMFindPackage(
    NAME KvikIO VERSION ${MIN_VERSION}
    GIT_REPOSITORY https://github.com/rapidsai/kvikio.git
    GIT_TAG branch-${MIN_VERSION}
    GIT_SHALLOW TRUE SOURCE_SUBDIR cpp
    OPTIONS "KvikIO_BUILD_EXAMPLES OFF"
  )
```

Reply:
Thanks, that worked :)